Sources/Persist/Persister+OSLog.swift

```diff
@@ -0,0 +1,30 @@
+#if canImport(os)
+import os.log
+import PersistLogger
+
+@available(macOS 10.12, iOS 10.0, watchOS 3.0, tvOS 10.0, *)
+extension Persister {
+    /**
+     Create a new `Persister` instance.
+
+     - parameter valueGetter: The closure that will be called when the `retrieveValue()` function is called.
+     - parameter valueSetter: The closure that will be called when the `persist(_:)` function is called.
+     - parameter valueRemover: The closure that will be called when the `removeValue()` function is called.
+     - parameter defaultValue: The value to use when a value has not yet been stored, or an error occurs. This value is lazily evaluated.
+     - parameter defaultValuePersistBehaviour: An option set that describes when to persist the default value. Defaults to `[]`.
+     - parameter addUpdateListener: A closure that will be called immediately to add an update listener.
+     */
+    public convenience init(
+        valueGetter: @escaping ValueGetter,
+        valueSetter: @escaping ValueSetter,
+        valueRemover: @escaping ValueRemover,
+        defaultValue: @autoclosure @escaping () -> Value,
+        defaultValuePersistBehaviour: DefaultValuePersistOption = [],
+        osLog: OSLog,
+        addUpdateListener: AddUpdateListener
+    ) {
+        let logger = OSLogPersistLogger(log: osLog)
```

Comment:
Just a thought on this.

I have recently found is beneficial to allow `OSLog? = nil` as a simple way to disable logging, particular in tests. The default value is subjective, but I prefer it personally so the default callsite is the API rather than additional features. Actually in my own use-cases I don't include it in init/function signatures as I feel it pollutes a little.

Reply:
`OSLog.disabled` does the same thing and removes the need for checking for `nil` so I prefer to use that in tests.

Reply:
Yeah, but that disables ALL logging through OSLog right? Outside of tests, that's less desirable.

Reply:
No it's a singleton `OSLog` instance that just discards whatever messages you sent to it.